### ISSUE TITLE
jsx-a11y/label-has-for: `<label>` can also be a containing element

### ISSUE THREAD

**Issue by alecf** [evcohen/eslint-plugin-jsx-a11y#84]
the `jsx-a11y/label-has-for` rule requires htmlFor=, but the `<label>` tag can also contain a form element:

```html
<label><input type="checkbox">Enabled</label>
```
Often this ends up being a more useful way to use label.

It would be nice if this rule would allow you to leave out htmlFor if it contained a form element.

See the label spec: https://html.spec.whatwg.org/multipage/forms.html#the-label-element
> If the for attribute is not specified, but the label element has a labelable element descendant, then the first such descendant in tree order is the label element's labeled control.

**Comment by ljharb** [comment]
Both nesting and for-ID linking are actually required to achieve 100% coverage on assistive devices.

**Comment by evcohen** [comment]
This is a dupe of #51 which we have a proposed solution for. Just reopened - lost track of it since it was closed 👍 thanks!
